eos-paygd/main.c

```diff
@@ -231,11 +233,36 @@ test_and_update_securitylevel (char *procname)
   return TRUE;
 }
 
+static gboolean
+secure_boot_enabled (void)
+{
+  uint8_t *secboot = NULL;
+  size_t data_size = 0;
+  uint32_t attributes;
+  int ret;
+
+  ret = efi_get_variable (EFI_GLOBAL_VARIABLE_GUID, "SecureBoot", &secboot, &data_size, &attributes);
+  if (ret < 0 || !secboot || data_size != 1)
```

Comment:
!secboot just means secure boot is off here, doesn't it? It should be left to the other conditional

Reply:
No, looking at the implementation in libefivar, I think it's just never the case that `ret == 0 && secboot == NULL`. And experimentally my machine has SB off and on it `secboot != NULL && *secboot == 0`. Side note: I changed the conditional below from `secboot == 0` to `*secboot == 0`.

Reply:
Oops, you're right, the other check was the bug, and you've fixed it.